test/common/upstream/upstream_impl_test.cc

```diff
@@ -568,24 +570,19 @@ TEST_P(StrictDnsClusterImplParamTest, DontWaitForDNSOnInit) {
 
   auto cluster = *createStrictDnsCluster(cluster_config, factory_context, dns_resolver_);
 
-  ReadyWatcher initialized;
+  MockFunction<absl::Status()> initialize_cb;
 
   // Initialized without completing DNS resolution.
-  EXPECT_CALL(initialized, ready());
-  cluster->initialize([&]() -> absl::Status {
-    initialized.ready();
-    return absl::OkStatus();
-  });
+  EXPECT_CALL(initialize_cb, Call).WillOnce(Return(absl::OkStatus()));
+  cluster->initialize(initialize_cb.AsStdFunction());
+  Mock::VerifyAndClearExpectations(&initialize_cb);
```

Comment:
Instead of this, would it accomplish the same to put the declaration of `initialize_cb` and this chunk of code into a `{}` scope?

Reply:
Makes sense, and done (and added comments in several places to make it clear what the scoping is for).

I've also done the rest of the file, removing all ReadyWatchers, which also removes a library dependency entirely, and made it a bit more readable with `using` typedefs.

Many of the ReadyWatchers in here weren't even getting used.